server.js

```diff
@@ -24,12 +26,87 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+const Book = mongoose.model("Book", {
+  bookID: String,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number,
+})
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany();
+    booksData.forEach((singleBook) => {
+      const newBook = new Book(singleBook);
+      newBook.save();
+    });
+  };
+  seedDatabase();
+}
+
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
+});
+
+app.get("/books/book/", async (req, res) => {
+  const {title, authors} = req.query;
+  const myRegex = /.*/gm
+
+  if (title && authors != undefined) {
+    const secondBook = await Book.find({
+      title: title ? title : myRegex,
+      authors: authors ? authors : myRegex})
+    res.send(secondBook);
+  } else {
+    const secondBook = await Book.find({
+      title: title})
+    res.send(secondBook);
+  }
+});
+
+//All books
+app.get('/booksData', async (req, res) => {
+  const allBooks = await Book.find()
+  res.send(allBooks)
+  })
+
+//Sort on title
+app.get("/books/book/:title", async (req, res) => {
+  const booksByTitle = await Book.find({title: req.params.title})
+  res.send(booksByTitle)
+});
+
+//Sort on authors
+app.get("/books/book/:authors", async (req, res) => {
+  const booksByAuthors = await Book.find({authors: req.params.authors})
+  res.send(booksByAuthors)
 });
 
+// Sorts out the book with the most amout of pages.
+app.get("/booksData/num_pages/max", async (req, res) => {
+  const booksByMaxPages = await Book.find().sort({num_pages:-1}).limit(1)
+  res.send(booksByMaxPages[0])
+})
+
+// Sorts out the book with the least amout of pages.
+app.get("/booksData/num_pages/min", async (req, res) => {
+  const booksByMinPages = await Book.find().sort({num_pages:+1}).limit(1)
+  res.send(booksByMinPages[0])
+})
+
+// Sorts out the book with the least amout of pages.
+app.get("/booksData/average_rating", async (req, res) => {
+  const booksByRating = await Book.find().sort( { average_rating:[-1] } )
+  res.send(booksByRating)
+})
+
```

Comment:
Unfortunately, I can't seem to get any data back from your deployed Heroku link. The API doesn't seem to work when I put in the endpoints you setup. I am wondering if it is working on your side locally because I don't seem to see anything from your deployed link. 

Did you add the environmental variables in a .env file of your project and add the environmental variables in Heroku?